app/experimenter/visualization/api/v3/views.py

```diff
@@ -112,55 +166,29 @@ def get_other_metrics_names_and_map(data, RESULTS_METRICS_MAP):
 def append_population_percentages(data):
     total_population = 0
     branches = {}
-    for row in data:
-        if row["metric"] == Metric.USER_COUNT:
-            total_population += row["point"]
-            branches[row["branch"]] = row["point"]
+    for jetstream_data_point in data:
+        if jetstream_data_point.metric == Metric.USER_COUNT:
+            total_population += jetstream_data_point.point
+            branches[jetstream_data_point.branch] = jetstream_data_point.point
 
     for branch_name, branch_user_count in sorted(branches.items()):
         data.append(
-            {
-                "metric": Metric.USER_COUNT,
-                "statistic": Statistic.PERCENT,
-                "branch": branch_name,
-                "point": round(branch_user_count / total_population * 100),
-            }
+            JetstreamDataPoint(
+                metric=Metric.USER_COUNT,
+                statistic=Statistic.PERCENT,
+                branch=branch_name,
+                point=round(branch_user_count / total_population * 100),
+            )
         )
```

Comment:
Is there a way to move all this parsing/transforming logic into more compact methods right on the pydantic models?